### PR TITLE
Added new zip action

### PR DESCRIPTION
**Pull request by KrauseFx**
This is requires with the `appetize` action

### CODE REVIEW

File: fastlane/fastlane.gemspec [hunk 1]
```diff
@@ -29,7 +29,6 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'terminal-table', '~> 1.4.5' # Actions documentation
   spec.add_dependency 'plist', '~> 3.1.0' # Needed for set_build_number_repository and get_info_plist_value actions
   spec.add_dependency 'addressable', '~> 2.3.8' # Support for URI templates
-
```
Comment (samrobbins) [inline fastlane/fastlane.gemspec]: Undo?

**Comment by samrobbins** [comment]
:truck: :dancer: :+1:

File: fastlane/fastlane.gemspec (hunk 1 above)
Comment (KrauseFx) [inline fastlane/fastlane.gemspec]: That's intended, since this space was unnecessary. :+1:

File: fastlane/lib/fastlane/actions/zip.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+module Fastlane
+  module Actions
+    class ZipAction < Action
+      def self.run(params)
+        UI.message "Compressing #{params[:path]}..."
+
+        params[:output_path] ||= "#{params[:path]}.zip"
+        sh "zip -r #{params[:output_path]} #{params[:path].shellescape}"
```
Comment (lacostej) [inline fastlane/lib/fastlane/actions/zip.rb]: @KrauseFx don't you need to shelleescape the `params[:output_path]` as well ?

File: fastlane/lib/fastlane/actions/zip.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Compress a file or folder to a zip"
+      end
+
+      def self.details
+        "This zips everything in a weird way if you're not in the directory of the file to zip"
```
Comment (mfurtak) [inline fastlane/lib/fastlane/actions/zip.rb]: If this behaves strangely when the action is not executed relative to the file or directory to be zipped, does it make sense to make the default behavior to `FileUtils.cd` to that location in a block around the `zip` command?

Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/zip.rb]: You mean `Dir.chdir`? If so, do we have to `cd` to the source or the destination?

File: fastlane/lib/fastlane/actions/zip.rb (hunk 2 above)
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/zip.rb]: Yeah, very good point, somehow missed htat

**Comment by KrauseFx** [comment]
I see this is already merged - I'll submit another PR addressing your 2 comments

**Pull request merged**